queue/driver_mongo.go

```diff
@@ -1191,108 +1191,6 @@ func (d *mongoDriver) JobInfo(ctx context.Context) <-chan amboy.JobInfo {
 	return infos
 }
 
-// getNextQuery returns the query for the next available jobs.
-func (d *mongoDriver) getNextQuery() bson.M {
-	now := time.Now()
-	staleBefore := now.Add(-d.LockTimeout())
-	qd := bson.M{
-		"$or": []bson.M{
-			d.getPendingQuery(bson.M{}),
-			d.getInProgQuery(
-				bson.M{"status.mod_ts": bson.M{"$lte": staleBefore}},
-			),
-		},
-	}
-
-	d.modifyQueryForGroup(qd)
-
-	var timeLimits []bson.M
-	if d.opts.CheckWaitUntil {
-		checkWaitUntil := bson.M{"$or": []bson.M{
-			{"time_info.wait_until": bson.M{"$lte": now}},
-			{"time_info.wait_until": bson.M{"$exists": false}},
-		}}
-		timeLimits = append(timeLimits, checkWaitUntil)
-	}
-
-	if d.opts.CheckDispatchBy {
-		checkDispatchBy := bson.M{"$or": []bson.M{
-			{"time_info.dispatch_by": bson.M{"$gt": now}},
-			{"time_info.dispatch_by": bson.M{"$exists": false}},
-		}}
-		timeLimits = append(timeLimits, checkDispatchBy)
-	}
-
-	if len(timeLimits) > 0 {
-		qd = bson.M{"$and": append(timeLimits, qd)}
-	}
-
-	return qd
-}
-
-// getNextSampledPipeline returns an aggregation pipeline to query for the next
-// available jobs, with a maximum limit (sampleSize) on the number of jobs
-// considered and returned. Jobs are returned in shuffled random order, and the
-// same job may be returned multiple times.
```

Comment:
Calling out: this removed the 
> and the same job may be returned multiple times

bit since I didn't understand how that would be.